### PR TITLE
Search perf improvements

### PR DESCRIPTION
**Pull request by andreamah**
Addresses #171436

Includes these improvements:
- schedules tree refreshes instead of only using `setInterval` with a set amount of time to reduce over-calling `refreshTree`
- optimize how we keep nodes collapsed/expanded using new collapses states from https://github.com/microsoft/vscode/issues/176505

### CODE REVIEW

File: src/vs/workbench/contrib/search/browser/searchView.ts [hunk 1]
```diff
@@ -1634,8 +1631,13 @@ export class SearchView extends ViewPane {
...
 					this.refreshAndUpdateCount();
 				}
-			}, 100);
-		}
+			} finally {
+				refreshUIScheduler.schedule();
+			}
+		};
+
+		const refreshUIScheduler = new RunOnceScheduler(refreshUI.bind(this), 80);
```
Comment (roblourens) [inline src/vs/workbench/contrib/search/browser/searchView.ts:1639]: This is disposable

File: src/vs/workbench/contrib/search/browser/searchView.ts [hunk 2]
```diff
@@ -251,6 +253,8 @@ export class SearchView extends ViewPane {
 
 		this.treeAccessibilityProvider = this.instantiationService.createInstance(SearchAccessibilityProvider, this.viewModel);
 		this.isTreeLayoutViewVisible = this.viewletState['view.treeLayout'] ?? (this.searchConfig.defaultViewMode === ViewMode.Tree);
+
+		this._refreshResultsScheduler = new RunOnceScheduler(this._updateResults.bind(this), 80);
```
Comment (roblourens) [inline src/vs/workbench/contrib/search/browser/searchView.ts:257]: You can use `this._register` too

**Review by roblourens — DISMISSED** [review]

File: src/vs/workbench/contrib/search/browser/searchView.ts [hunk 3]
```diff
@@ -1484,6 +1485,27 @@ export class SearchView extends ViewPane {
 			.then(() => undefined, () => undefined);
 	}
 
+
+	private _updateResults() {
+		try {
+			if (this.state === SearchUIState.Idle) {
+				window.clearInterval(this._uiRefreshHandle);
```
Comment (rzhao271) [inline src/vs/workbench/contrib/search/browser/searchView.ts:1492]: Is the handle still needed? I'm not sure where it's being set or used.

Comment (andreamah) [reply src/vs/workbench/contrib/search/browser/searchView.ts:1492]: good catch, it isn't used anymore

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**